pkg/process/util/api/weighted_queue.go

```diff
@@ -168,12 +162,17 @@ func (q *WeightedQueue) Add(item WeightedItem) {
 
 	q.queue.PushBack(item)
 
-	// Send a signal on the dataAvailable channel if needed
-	select {
-	case q.dataAvailable <- struct{}{}:
-	default:
+	// Send a signal that data is available
+	q.cv.Signal()
+}
 
-	}
+// Stop stops the WeightedQueue instance.  Any calls to Poll concurrent with or
+// after the call to Stop will return (nil, false) immediately.
+func (q *WeightedQueue) Stop() {
+	q.mu.Lock()
+	q.stop = true
+	q.cv.Signal()
```

Comment:
If there are multiple threads waiting on `Poll` won't we wake up only one of them in case we stop the queue? Don't we need to use a `Broadcast` here instead?

Reply:
Yes!  "I was just testing you"? :)